package.json

```diff
@@ -472,50 +468,64 @@
         "title": "Open Documentation",
         "icon": "$(book)"
       },
+      {
+        "command": "terraform.cloud.login",
+        "title": "HashiCorp Terraform Cloud: Login",
+        "enablement": "terraform.cloud.signed-in === false"
```

Comment:
Wouldn't users who end up with expired token still want to re-login? Or do we log them out automatically in that case, so this is still relevant? Or do we prompt them from the point we detect expired token automatically and we can just rely on that?

Probably not a big deal, just a question that came to mind!

Reply:
When there is an authentication error like a token expiry, we call `handleAuthArror()`, which directly calls the vscode authentication API to create a new session. That logs out the user and pops the modal dialog to log back in. At this point they're not signed in and the other commands disappear

Reply:
👍🏻 That's good enough, thanks for the explanation.